config/config-builder/src/full_node_config.rs

```diff
@@ -166,6 +168,7 @@ impl FullNodeConfig {
         let new_net = full_node_config.full_node_networks.swap_remove(0);
         for net in &config.full_node_networks {
             ensure!(new_net.peer_id != net.peer_id, "Network already exists");
+            // TODO do we want to distinguish between public network ids
```

Comment:
There is only one public network, in my opinion :D.

Reply:
If public==unauthenticated then yes, but there may also be an authenticated fullnode-only network for trusted external fullnodes to use (e.g. fullnodes run by Novi in FB infra connecting to the fullnode of Novi's validator deployment).

Reply:
TL;DR --- the term _public_ is overloaded and _public network_ means different things to different people.

TL;DR(2) --- A clear, concise, consistent understanding of deployment targets is needed.


A couple of different definitions of _public_

* unauthenticated and open to full participation by anyone
* accessible to anyone

The former definition is most inline with traditional blockchain projects.  This definition goes hand in hand with unconstrained deployments.

The second definition allows for networks which are deployed in a controlled environment, but accessible to anyone. A reasonable analogue is the public front ends that stand between users  and the back end infrastructure.

Looking at the [discussion](https://github.com/libra/libra/issues/3985#issuecomment-636659952) in #3985, one irony is that the suggested "private" networks connected to the public ring are in fact the most inline with the first definition of public.